sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -323,6 +341,18 @@ public String getId() {
     public <T> Mono<CosmosItemResponse<T>> upsertItem(T item, PartitionKey partitionKey, CosmosItemRequestOptions options) {
         final CosmosItemRequestOptions requestOptions = options == null ? new CosmosItemRequestOptions() : options;
         ModelBridgeInternal.setPartitionKey(requestOptions, partitionKey);
+
+        if (this.bulkExecutionEnabled) {
```

Comment:
I would push the code in the if this.bulkExecutionEnabled  for item operations condition into its own private method (and passing in teh operationType) to reduce code duplication

Reply:
we are doing same thing right? Passing operation type and leaving remaining for bulkExecutionInternal. We have to return it inside if condition here.